File: backend/sit_igp_management_backend/api/professors/professors_service.py
from typing import Any

from sit_igp_management_backend.api.professors.entities.professor import Professor


def create(professor: Professor) -> Any:
    return {
        "full_name": professor.full_name,
        "email": professor.email,
    }


def find_all() -> Any:
    return None


def find_one(professor_id: int) -> Any:
    return {
        "professor_id": professor_id,
    }


def update(professor_id: int, update_professor_dto: Professor) -> Any:
    return {
        "professor_id": professor_id,
        "full_name": update_professor_dto.full_name,
        "email": update_professor_dto.email,
    }


def remove(professor_id: int) -> Any:
    return {
        "professor_id": professor_id,
    }

File: frontend/core/api/professors_service.ts
import Professor from '../types/professor';
import { apiClient } from './apiClient';

class ProfessorsService {
  async create(obj: Professor) {
    const response = await apiClient.post<Professor>('/professors/', obj);
    return response.data;
  }

  async findAll() {
    const response = await apiClient.get<Professor[]>('/professors/');
    return response.data;
  }

  async findById(professor_id: string) {
    const response = await apiClient.get<Professor | null>(
      `/professors/${professor_id}`
    );
    return response.data;
  }

  async update(professor_id: string, obj: Professor) {
    const response = await apiClient.put<Professor>(
      `/professors/${professor_id}`,
      obj
    );
    return response.data;
  }

  async delete(professor_id: string) {
    await apiClient.delete(`/professors/${professor_id}`);
  }
}

const professors_service = new ProfessorsService();

export default professors_service;
